### PR TITLE
fix(status): keep rotated heatmap column labels clear of the cell grid

### LINKED ISSUE #1518: Replication heatmap: rotated column axis labels are hidden behind the top row of cells
Noticed while reviewing the replication-latency heatmap in #1508. **Pre-existing** in `HeatmapMatrix` — #1508 doesn't touch the header rendering, but by making the heatmap interactive (cell drilldown) it puts the flaw front and center.

### Symptom
The rotated column (destination) axis labels at the top of the heatmap are partly **hidden behind the top row of cells** — the lower end of each diagonal label disappears under the cells.

### Cause
In [`HeatmapMatrix.tsx`](src/features/instance/status/analytics/primitives/HeatmapMatrix.tsx#L506) the column labels are:
- anchored at `y = HEADER_HEIGHT - 8` = **64px**, only **8px above** the cell grid (cells start at `HEADER_HEIGHT` = 72px);
- drawn with `textAnchor="end"` + `transform="rotate(-45, cx, cy)"`.

That rotation sends the text **down-and-left** from the anchor, so the far end of an 8–20-char label lands ~30–45px *below* the anchor — inside the first cell row. And the header `<g role="row">` is painted **before** the data-row cells, so the cells paint over the overlapping portion. Net: labels tucked under the cells.

### Possible directions (triage as you see fit, @kylebernhardy)
- Give the labels room so they never enter the cell region: increase `HEADER_HEIGHT` and/or reposition the anchor so the label's *lowest* point sits at the header bottom (not its baseline).
- And/or flip the geometry so the diagonal text extends **up**-left (away from the grid) rather than down-left.
- Rendering the header row after the cells would un-hide them, but they'd then overlap the cell fills — so the real fix is geometry/space, not just z-order.

Cosmetic, non-blocking. Affects every heatmap (currently just replication latency).

### PR DESCRIPTION
**Pull request by kylebernhardy**
## Symptom

In the replication-latency heatmap (`HeatmapMatrix`), the rotated destination (column) axis labels at the top were partly **hidden behind the first row of cells** — the lower end of each diagonal label disappeared under the cells. Pre-existing, surfaced by the interactive drilldown in #1508.

## Cause

The column labels anchored at `y = HEADER_HEIGHT - 8` (only 8px above the grid) and rendered with `textAnchor="end"` + `transform="rotate(-45, cx, cy)"`. The `-45` rotation sends the far end **down-and-left** — ~30-45px below the anchor for an 8-20 char label, i.e. inside the first cell row. And the header `<g role="row">` paints **before** the data rows, so the cells paint over the overlap.

## The geometry change

Make a label unable to enter the cell region, rather than masking with z-order:

- **Flip the rotation to `+45`** (still `textAnchor="end"`) so labels lean **up-and-left**, away from the grid. The anchored END glyph becomes the label's **lowest** point, so its lowest extent is fixed at the anchor baseline **regardless of label length** — extra length only lifts the far (top) end, never pushes it down into the cells.
- **Anchor that near end just above the first cell row** (`headerHeight - HEADER_BOTTOM_PAD`).
- **Replace the fixed `HEADER_HEIGHT = 72` constant with `computeHeaderHeight(cols, cellSize)`** — it sizes the header to the longest post-truncation label's diagonal vertical extent (`width·sinθ + fontSize·cosθ`), floored at 72px, so the up-left far end is never clipped at `y=0`. `svgHeight`/`viewBox` derive from it, so the taller header is always contained (no `height` prop is passed in production).

## Where to focus review

- The anchor math and rotation direction (`primitives/HeatmapMatrix.tsx`, the column-header `<text>` and `computeHeaderHeight`).
- `computeHeaderHeight` uses a per-character width estimate (`COL_LABEL_CHAR_W`) rather than DOM text measurement, to stay pure/deterministic (SSR + tests). A pathological all-caps/CJK label could still exceed the estimate, but because of the near-end anchoring that only lifts the label's **top** (never its bottom into the cells), and `overflow: visible` renders any excess rather than clipping. Char width was set to the generous side of a mixed-case hostname average.

## Tests

Extends the `HeatmapMatrix` suite with a column-label geometry block (labels anchored above the first cell row; `+45` not `-45`; first cell row starts exactly at `computeHeaderHeight`; viewBox contains the header) and unit tests for `computeHeaderHeight` (floor at 72, growth for long labels, monotonicity, per-`cellSize` truncation cap). Full suite green (1566 passed / 11 skipped); tsc / dprint / oxlint clean.

## Review adjudication

- **Codex** [P2]: the per-char width estimate can under-reserve for wide glyphs. Addressed — bumped the estimate and documented the bound + `overflow: visible` safety net (cell-overlap fix is width-independent).
- **Gemini**: no issues; validated the geometry, header sizing, edge cases (1 col / empty cols / long labels), and that the tests pin the invariant.

Fixes #1518

Comment generated by kAIle (Claude Fable 5)

### CODE REVIEW

File: src/features/instance/status/analytics/primitives/HeatmapMatrix.tsx [hunk 1]
```diff
@@ -242,9 +242,54 @@ function HeatmapColorLegend({ stops, vmin, vmax, width, axis, approx, measureLab
...
+ *  in tests and SSR. */
+export function computeHeaderHeight(cols: string[], cellSize: number): number {
+	const maxChars = cols.reduce(
+		(m, c) => Math.max(m, Math.min(c.length, colTruncateLength(cellSize))),
+		0,
+	);
+	const rad = (COL_LABEL_ANGLE_DEG * Math.PI) / 180;
+	const vExtent = maxChars * COL_LABEL_CHAR_W * Math.sin(rad) + COL_LABEL_FONT_SIZE * Math.cos(rad);
+	return Math.max(MIN_HEADER_HEIGHT, Math.ceil(vExtent) + HEADER_TOP_PAD + HEADER_BOTTOM_PAD);
+}
```
[the remark this answers is not in the document]
Comment (kylebernhardy) [reply src/features/instance/status/analytics/primitives/HeatmapMatrix.tsx:294]: Keeping the explicit sin/cos here. It's evaluated once per render (not per cell), so the trig cost is negligible, and reading `Math.sin(rad)/Math.cos(rad)` against `COL_LABEL_ANGLE_DEG` keeps the geometry self-documenting and correct if the angle is ever tuned — hardcoding `SQRT1_2` would silently produce a wrong header height the moment the angle stops being 45°.

Comment generated by kAIle (Claude Fable 5)

**Review by dawsontoth — APPROVED** [review]

File: src/features/instance/status/analytics/primitives/HeatmapMatrix.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (kylebernhardy) [reply src/features/instance/status/analytics/primitives/HeatmapMatrix.tsx:294]: Applied after all in a follow-up commit — factored to `Math.SQRT1_2` with a comment documenting that it assumes the fixed 45° angle (so a future angle change restores the general sin/cos form). Thanks for the nudge.

Comment generated by kAIle (Claude Fable 5)

**Review by kriszyp — APPROVED** [review]
LGTM. Keeps rotated heatmap column labels clear of the cell grid; consistent with the rest of the status/analytics chart series.

— review via KrAIs (Claude) on Kris's behalf
